### PR TITLE
Use matplotlib's infinite axline to demonstrate hough transform

### PR DESCRIPTION
**Pull request by stefanv**
This requires matplotlib 3.3.

It does prevent problems when `np.sin(angle)` is zero.

### CODE REVIEW

**Comment by grlee77** [comment]
I think this actually needs Matplotib >= 3.3!. When I tried with 3.1.1 I got `AttributeError: 'AxesSubplot' object has no attribute 'axline'`

There are `axhline` and `axvline` in 3.0, but not `axline`.

**Comment by grlee77** [comment]
Matplotlib 3.0 is nearly two years old, so that seems reasonable to require. Matplotlib 3.3, though, was just released 15 days ago! Do you think it is worth having two code paths in the example until we make 3.3 a requirement or should we hold off on this change for now?

**Comment by tacaswell** [comment]
Yeah, this is a 3.3 feature.

**Comment by stefanv** [comment]
Woops, meant to write 3.3 ;)

**Comment by stefanv** [comment]
I think using Arch linux has messed with me.

What?! Not running latest? Get at it!

**Comment by hmaarrfk** [comment]
Can we change the dependency for the docs instead?
https://github.com/scikit-image/scikit-image/blob/master/requirements/docs.txt

We already require a newer version of matplotlib for docs than for the build.

**Comment by emmanuelle** [comment]
I agree with Mark here: if we bump a dependency for a version which is only 2 weeks old, it should be the docs dependencies, not the package ones.

**Comment by stefanv** [comment]
Sure, that's fine. But I may hold off until 3.3 is widely available to users, otherwise this won't run for most readers.

File: requirements/default.txt
Comment (sciunto) [inline requirements/default.txt:3]:
```suggestion
matplotlib>=3.3
```

**Comment by madphysicist** [comment]
FWIW, I'm all for this change.

File: requirements/default.txt
Comment (sciunto) [reply requirements/default.txt:3]: Would you accept this change and would you like to rebase @stefanv ?

I'm OK to bump for a 6 months old version.

**Review by sciunto — APPROVED** [review]

**Comment by jni** [comment]
I'm happy to merge ASAP if we change this to happen with the docs requirement rather than the default ones.

**Comment by stefanv** [comment]
@rfezzani I was wondering if you'd mind taking a look at this one.

**Comment by stefanv** [comment]
OK, should be good to go.

File: requirements/default.txt
Comment (rfezzani) [inline requirements/default.txt:3]: Is this change intentional?

**Review by rfezzani — APPROVED** [review]
I just wonder if you meant matplotlib version decrease, otherwise I am :+1: for the introduced changes. Thank you @stefanv

File: requirements/default.txt
Comment (stefanv) [reply requirements/default.txt]: Woops, no, thanks for catching that; fixed.

**Pull request merged**
